extension.js

```diff
@@ -58,15 +57,17 @@ var ContainersMenu = GObject.registerClass(
 
             hbox.add_child(icon);
             this.add_child(hbox);
-            this.connect("button_press_event", () => {
+
+            this.menu.connect("open-state-changed", () => {
```

Comment:
can't seem to find the docs about the difference between button_press_event to open-state-changed

Reply:
`button-press-event` corresponds to a raw hardware event (here: a pointer event), and is defined by the `ClutterActor` parent class.

`open-state-changed` is specific to `PopupMenu`, and is emitted when the menu is opened or closed, regardless of the event that triggered the action.

The last bit is also the most notable difference: The current code only refreshes the menu if it was opened by a click, but not if it was done via keyboard or programmatically.